### PR TITLE
Data types overview task

### PR DESCRIPTION
**Pull request by andreamamchur**

### CODE REVIEW

File: Exercises/data-types-overview/data-types-overview.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+const typesCount = {
+  number : 0,
+  string: 0,
+  boolean: 0,
+  object: 0,
+  undefined: 0 };
+
+for (let i = 0; i < array.length; i++) {
+  typesCount[typeof array[i]] += 1;
+}
```
Comment (Mike-Savin) [inline Exercises/data-types-overview/data-types-overview.js]: Try to rewrite the code to fill empty object, not prefilled with zero values.

File: Exercises/data-types-overview/data-types-overview.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+//number
+const number = 123;
+const number2 = Number('456');
+//string
+const string = 'some string';
+const string2 = String('new string');
+
+//boolean
+const booleanVariable = true;
+const booleanVariable2 = Boolean(false);
```
Comment (Mike-Savin) [inline Exercises/data-types-overview/data-types-overview.js]: When you use `Number/String/Boolean` without `new` keyword before it is not a constructor but just a function. You can read some facts about the difference here:
https://stackoverflow.com/a/4719342
In this task constructors should be used, not functions.

**Review by Mike-Savin — CHANGES_REQUESTED** [review]

File: Exercises/data-types-overview/data-types-overview.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+
+//undefined
+let undefinedVariable;
+
+//null
+const nullVariable = null;
+
+//object
+const object = {
```
Comment (lizarusi) [inline Exercises/data-types-overview/data-types-overview.js:24]: you can create object with constructor as well

File: Exercises/data-types-overview/data-types-overview.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    string: 'some text',
+    'long name': undefined,
+    nullVariable: null,
+    booleanVariable: false,
+    object2: { a: 123, b: 'text' }
+};
+
+const objectWithReferencesNew = objectWithReferences;
+
+objectWithReferences.number = 123.123;
```
Comment (lizarusi) [inline Exercises/data-types-overview/data-types-overview.js:57]: 👍

File: Exercises/data-types-overview/data-types-overview.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+// Declare an array containing different data types.
+
+const array = [123, 'string', false, null, undefined, { a: 1 }];
+
+// Iterate through it and collect amount of each data type to object.
+
+const typesCount = {};
+
+for (let i = 0; i < array.length; i++) {
+    if (typesCount[typeof array[i]] === undefined) {
```
Comment (lizarusi) [inline Exercises/data-types-overview/data-types-overview.js]: great! now you should know logical operators and can simplify it :)

**Review by lizarusi — CHANGES_REQUESTED** [review]

**Review by Mike-Savin — APPROVED** [review]

**Pull request merged**